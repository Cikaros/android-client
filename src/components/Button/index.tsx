import React from 'react';
import { Text, TouchableOpacity, View, ViewStyle, FlexAlignType, Dimensions } from 'react-native';
import Lottie from 'lottie-react-native';

const windowWidth = Dimensions?.get('window')?.width;

interface Props {
  text: string;
  id?: string;
  outline?: boolean;
  loading?: boolean;
  imagePrefix?: () => JSX.Element;
  onPress: () => void;
}

const buttonStyles = {
  touchableOpacity: {
    borderWidth: 1,
    borderColor: '#f35e32',
    borderRadius: 2,
    padding: 8,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#f68330',
  } as ViewStyle,
  touchableOpacityOutline: {
    borderWidth: 1,
    borderColor: '#f35e32',
    borderRadius: 2,
    padding: 8,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    
  } as ViewStyle,
  content: {
    width: (windowWidth * 0.7),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  } as ViewStyle,
};

const Button: React.FC<Props> = ({
  text,
  outline,
  loading,
  imagePrefix,
  onPress,
}) => {

  const showLoading = (loading: boolean | undefined) => {
    if (loading) {
      return (
        <View style={{ width: 50, height: '100%' }}>
          <Lottie
            source={require('../../assets/animations/loading.json')}
            autoPlay
            loop
          />
        </View>
      );
    }
    return null;
  };

  return (
    <>
      {!outline ? (
        <TouchableOpacity onPress={onPress}>
          <View style={buttonStyles.touchableOpacity}>
            <View style={buttonStyles.content}>
              {imagePrefix && imagePrefix()}
              <Text style={{ fontSize: 14, lineHeight: 22, color: '#ffffff' }}>
                {text}
              </Text>
              {showLoading(loading)}
            </View>
          </View>
        </TouchableOpacity>
      ) : (
        <TouchableOpacity onPress={onPress}>
          <View style={buttonStyles.touchableOpacityOutline}>
            {imagePrefix && imagePrefix()}
            <Text style={{ fontSize: 14, lineHeight: 22, color: '#F35E32' }}>{text}</Text>
          </View>
        </TouchableOpacity>
      )}
    </>
  );
};

export default React.memo(Button);
