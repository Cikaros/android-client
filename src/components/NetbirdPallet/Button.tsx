import styled from 'styled-components';
import {Text, TouchableOpacity, View} from 'react-native';

export const NbTouchableOpacity = styled(TouchableOpacity)`
  border: 1px solid #f35e32;
  border-radius: 2px;
  padding: 8px 15px;
  align-self: stretch;
  align-items: center;
  justify-items: center;
  background-color: #f68330;
`;

export const NbTextTouchableOpacity = styled(Text)`
  font-size: 14px;
  line-height: 22px;
  color: #ffffff;
`;

export const NbTouchableOpacityOutline = styled(NbTouchableOpacity)`
  background-color: transparent;
  border-color: #f35e32;
`;

export const NbTextTouchableOpacityOutline = styled(NbTextTouchableOpacity)`
  color: #f68330;
`;

export const NbContent = styled(View)`
  flex-direction: row;
  align-items: center;
`;
