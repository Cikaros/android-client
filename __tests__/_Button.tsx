import React from 'react';
import { shallow } from 'enzyme';
import Button from '../src/components/Button/index';

describe('Button', () => {
  const mockProps = {
    text: 'Click me',
    outline: false,
    loading: false,
    imagePrefix: jest.fn(),
    onPress: jest.fn(),
  };

  it('renders correctly without outline and loading', () => {
    const wrapper = shallow(<Button {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with outline', () => {
    const wrapper = shallow(<Button {...mockProps} outline={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with loading', () => {
    const wrapper = shallow(<Button {...mockProps} loading={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders image prefix when imagePrefix prop is provided', () => {
    const wrapper = shallow(<Button {...mockProps} imagePrefix={jest.fn()} />);
    expect(wrapper.find('imagePrefix')).toBeTruthy();
  });
});
