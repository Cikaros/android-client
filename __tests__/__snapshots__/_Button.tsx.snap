// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly with loading 1`] = `
<Fragment>
  <Component
    onPress={[MockFunction]}
  >
    <Component
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f68330",
          "borderColor": "#f35e32",
          "borderRadius": 2,
          "borderWidth": 1,
          "justifyContent": "center",
          "padding": 8,
          "width": "70%",
        }
      }
    >
      <Component
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": NaN,
          }
        }
      >
        <Component
          style={
            {
              "color": "#ffffff",
              "fontSize": 14,
              "lineHeight": 22,
            }
          }
        >
          Click me
        </Component>
        <Component
          style={
            {
              "height": "100%",
              "width": 50,
            }
          }
        >
          <mockConstructor
            autoPlay={true}
            loop={true}
            source={
              {
                "assets": [],
                "ddd": 0,
                "fr": 29.9700012207031,
                "h": 24,
                "ip": 0,
                "layers": [
                  {
                    "ao": 0,
                    "bm": 0,
                    "ddd": 0,
                    "ind": 1,
                    "ip": 0,
                    "ks": {
                      "a": {
                        "a": 0,
                        "ix": 1,
                        "k": [
                          0,
                          0,
                          0,
                        ],
                        "l": 2,
                      },
                      "o": {
                        "a": 0,
                        "ix": 11,
                        "k": 100,
                      },
                      "p": {
                        "a": 0,
                        "ix": 2,
                        "k": [
                          12,
                          12,
                          0,
                        ],
                        "l": 2,
                      },
                      "r": {
                        "a": 0,
                        "ix": 10,
                        "k": 0,
                      },
                      "s": {
                        "a": 0,
                        "ix": 6,
                        "k": [
                          100,
                          100,
                          100,
                        ],
                        "l": 2,
                      },
                    },
                    "nm": "Shape Layer 1",
                    "op": 120.0000048877,
                    "shapes": [
                      {
                        "bm": 0,
                        "cix": 2,
                        "hd": false,
                        "it": [
                          {
                            "d": 1,
                            "hd": false,
                            "mn": "ADBE Vector Shape - Ellipse",
                            "nm": "Ellipse Path 1",
                            "p": {
                              "a": 0,
                              "ix": 3,
                              "k": [
                                0,
                                0,
                              ],
                            },
                            "s": {
                              "a": 0,
                              "ix": 2,
                              "k": [
                                24,
                                24,
                              ],
                            },
                            "ty": "el",
                          },
                          {
                            "bm": 0,
                            "c": {
                              "a": 0,
                              "ix": 3,
                              "k": [
                                1,
                                1,
                                1,
                                1,
                              ],
                            },
                            "hd": false,
                            "lc": 1,
                            "lj": 1,
                            "ml": 4,
                            "mn": "ADBE Vector Graphic - Stroke",
                            "nm": "Stroke 1",
                            "o": {
                              "a": 0,
                              "ix": 4,
                              "k": 100,
                            },
                            "ty": "st",
                            "w": {
                              "a": 0,
                              "ix": 5,
                              "k": 2,
                            },
                          },
                          {
                            "a": {
                              "a": 0,
                              "ix": 1,
                              "k": [
                                0,
                                0,
                              ],
                            },
                            "nm": "Transform",
                            "o": {
                              "a": 0,
                              "ix": 7,
                              "k": 100,
                            },
                            "p": {
                              "a": 0,
                              "ix": 2,
                              "k": [
                                0,
                                0,
                              ],
                            },
                            "r": {
                              "a": 0,
                              "ix": 6,
                              "k": 0,
                            },
                            "s": {
                              "a": 0,
                              "ix": 3,
                              "k": [
                                75.141,
                                75.141,
                              ],
                            },
                            "sa": {
                              "a": 0,
                              "ix": 5,
                              "k": 0,
                            },
                            "sk": {
                              "a": 0,
                              "ix": 4,
                              "k": 0,
                            },
                            "ty": "tr",
                          },
                        ],
                        "ix": 1,
                        "mn": "ADBE Vector Group",
                        "nm": "Ellipse 1",
                        "np": 2,
                        "ty": "gr",
                      },
                      {
                        "e": {
                          "a": 1,
                          "ix": 2,
                          "k": [
                            {
                              "e": [
                                100,
                              ],
                              "i": {
                                "x": [
                                  0.667,
                                ],
                                "y": [
                                  1,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.333,
                                ],
                                "y": [
                                  0,
                                ],
                              },
                              "s": [
                                0,
                              ],
                              "t": 0,
                            },
                            {
                              "t": 104.000004236007,
                            },
                          ],
                        },
                        "hd": false,
                        "ix": 2,
                        "m": 1,
                        "mn": "ADBE Vector Filter - Trim",
                        "nm": "Trim Paths 1",
                        "o": {
                          "a": 1,
                          "ix": 3,
                          "k": [
                            {
                              "e": [
                                720,
                              ],
                              "i": {
                                "x": [
                                  0.833,
                                ],
                                "y": [
                                  0.833,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.167,
                                ],
                                "y": [
                                  0.167,
                                ],
                              },
                              "s": [
                                0,
                              ],
                              "t": 0,
                            },
                            {
                              "t": 119.000004846969,
                            },
                          ],
                        },
                        "s": {
                          "a": 1,
                          "ix": 1,
                          "k": [
                            {
                              "e": [
                                100,
                              ],
                              "i": {
                                "x": [
                                  0.533,
                                ],
                                "y": [
                                  1,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.387,
                                ],
                                "y": [
                                  0,
                                ],
                              },
                              "s": [
                                0,
                              ],
                              "t": 25,
                            },
                            {
                              "t": 118.000004806239,
                            },
                          ],
                        },
                        "ty": "tm",
                      },
                    ],
                    "sr": 1,
                    "st": 0,
                    "ty": 4,
                  },
                ],
                "markers": [],
                "nm": "loading",
                "op": 120.0000048877,
                "v": "5.9.0",
                "w": 24,
              }
            }
          />
        </Component>
      </Component>
    </Component>
  </Component>
</Fragment>
`;

exports[`Button renders correctly with outline 1`] = `
<Fragment>
  <Component
    onPress={[MockFunction]}
  >
    <Component
      style={
        {
          "alignItems": "center",
          "borderColor": "#f35e32",
          "borderRadius": 2,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 8,
        }
      }
    >
      <Component
        style={
          {
            "color": "#F35E32",
            "fontSize": 14,
            "lineHeight": 22,
          }
        }
      >
        Click me
      </Component>
    </Component>
  </Component>
</Fragment>
`;

exports[`Button renders correctly without outline and loading 1`] = `
<Fragment>
  <Component
    onPress={[MockFunction]}
  >
    <Component
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f68330",
          "borderColor": "#f35e32",
          "borderRadius": 2,
          "borderWidth": 1,
          "justifyContent": "center",
          "padding": 8,
          "width": "70%",
        }
      }
    >
      <Component
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "width": NaN,
          }
        }
      >
        <Component
          style={
            {
              "color": "#ffffff",
              "fontSize": 14,
              "lineHeight": 22,
            }
          }
        >
          Click me
        </Component>
      </Component>
    </Component>
  </Component>
</Fragment>
`;
